lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -849,7 +851,45 @@ def test_student_progress_with_valid_and_invalid_id(self, default_store):
         # Assert that valid 'student_id' returns 200 status
         self.assertEqual(resp.status_code, 200)
 
-    def test_non_asci_grade_cutoffs(self):
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_unenrolled_student_progress_for_credit_course(self, default_store):
+        """
+         Test that student progress page does not break while checking for an unenrolled student.
+
+         Scenario: When instructor checks the progress of a student who is not enrolled in credit course.
+         It should return 200 response.
+        """
+        # Create a new course, a user which will not be enrolled in course, admin user for staff access
+        course = CourseFactory.create(default_store=default_store)
+        not_enrolled_user = UserFactory.create()
+        self.request.user = AdminFactory.create()
+
+        # Create and enable Credit course
+        CreditCourse.objects.create(course_key=course.id, enabled=True)
+
+        # Configure a credit provider for the course
+        CreditProvider.objects.create(
+            provider_id="ASU",
+            enable_integration=True,
+            provider_url="https://credit.example.com/request",
+        )
+
+        requirements = [
+            {
+                "namespace": "grade",
+                "name": "grade",
+                "display_name": "Grade",
+                "criteria": {"min_grade": 0.52},
+            }
+        ]
+        # Add a single credit requirement (final grade)
+        set_credit_requirements(course.id, requirements)
+
+        resp = views.progress(self.request, course_id=course.id.to_deprecated_string(),
```

Comment:
Hey @Ayub-Khan , the guidelines are looking for something like:
```python
resp = views.progress(
    self.request, course_id=unicode(course.id), student_id=not_enrolled_user.id
)
```

or
```python
resp = views.progress(
    self.request,
    course_id=unicode(course.id),
    student_id=not_enrolled_user.id,
)
```

Also, do `unicode(course.id)` instead of `course.id.to_deprecated_string()`